tests/phpunit/tests/media/filesize.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+/**
+ * @group media
+ * @group media_filesize
+ */
+class Tests_Image_Filesize extends WP_UnitTestCase {
+	function tear_down() {
+		$this->remove_added_uploads();
+
+		parent::tear_down();
+	}
+
+	/**
+	 * Check that filesize meta is generated for jpegs.
+	 *
+	 * @ticket 49412
+	 */
+	function test_filesize_in_jpg_meta() {
+		$attachment = $this->factory->attachment->create_upload_object( DIR_TESTDATA . '/images/33772.jpg' );
+
+		$metadata = wp_get_attachment_metadata( $attachment );
+
+		$this->assertEquals( wp_filesize( get_attached_file( $attachment ) ), $metadata['filesize'] );
+
+		foreach ( $metadata['sizes'] as $intermediate_size ) {
+			$this->assertTrue( ! empty( $intermediate_size['filesize'] ) && is_numeric( $intermediate_size['filesize'] ) );
```

Comment:
I think being more expressive here would be ideal, and subsequent assertions.

```suggestion
			$this->assertArrayHasKey( 'filesize', $intermediate_size );
			$this->assertNotEmpty( $intermediate_size['filesize'] );
			$this->assertIsNumeric( $intermediate_size['filesize'] );
```